code/game/objects/items/weapons/storage/boxes.dm

```diff
@@ -664,3 +664,100 @@
 #undef SYNDI
 #undef HEART
 #undef SMILE
+
+/obj/item/weapon/storage/box/ingredients //This box is for the randomely chosen version the chef spawns with, it shouldn't actually exist.
+	name = "ingredients box"
+	icon_state = "donk_kit"
+	item_state = null
+
+/obj/item/weapon/storage/box/ingredients/wildcard
+	item_state = "wildcard"
+
+/obj/item/weapon/storage/box/ingredients/fiesta
+	item_state = "fiesta"
+
+/obj/item/weapon/storage/box/ingredients/italian
+	item_state = "italian"
+
+/obj/item/weapon/storage/box/ingredients/vegetarian
+	item_state = "vegetarian"
+
+/obj/item/weapon/storage/box/ingredients/sweets
+	item_state = "sweets"
+
+/obj/item/weapon/storage/box/ingredients/carnivore
+	item_state = "carnivore"
+
+/obj/item/weapon/storage/box/ingredients/exotic
+	item_state = "exotic"
+
+/obj/item/weapon/storage/box/ingredients/New()
+	..()
+	if (item_state == null)
+		item_state = pick("wildcard", "fiesta", "italian", "vegetarian", "sweets", "carnivore", "exotic")
+	desc = "A box containing supplementary ingredients for the aspiring chef. This box's theme is '[item_state]'."
+	switch(item_state)

```

Comment:
you should offload this whole switch statement to a new proc, and override that proc in the children.